system/security/shared/caching.cpp

```diff
@@ -210,7 +210,10 @@ void CResPermissionsCache::remove(SecResourceType rtype, const char* resourcenam
 
 CPermissionsCache::~CPermissionsCache()
 {
-    g_mapCache.clear();
+    {
+        CriticalBlock block(PCCritSect);
+        g_mapCache.erase(m_secMgrClass.str());
```

Comment:
Interesting question. In everything I read about map::erase nothing is mentioned about null. I tested with null and it worked, but it's probably up to the compiler implementation of the compare function. Just to be safe, I will test for NULL before performing the delete